cdap-docs/developers-manual/source/pipelines/developing-plugins/creating-a-plugin.rst

```diff
@@ -95,6 +95,14 @@ the types, only one method is required to be implemented::
 - ``transform()``: This method will be called for every input key-value pair generated by
   the batch job. By default, the value is emitted to the subsequent stage.
 
+.. rubric:: Lineage
+
+For plugins that fetch data from non-CDAP sources, the lineage is registered using the ``inputName`` provided
+when ``setInput()`` is invoked on ``BatchSourceContext`` in ``prepareRun()``. Note that the ``inputName`` should
+be a valid ``DatasetId``. Optionally, you can extend from ``ReferenceBatchSource``, which verifies the
```

Comment:
I think if we want to do anything, we should put ReferencePluginConfig in hydrator api with a validate() method that checks the referenceName is valid, rather than having a ReferenceBatchSource. Nothing else in ReferenceBatchSource is actually doing anything related to lineage. Children of that class still have to use the reference name when calling setInput().